setup.py

```diff
@@ -0,0 +1,204 @@
+    elif not sha and returncode == 129:
+        # git returns 129 if a command option failed to parse; in
+        # particular this could happen in git versions older than 1.7.2
+        # where the --count option is not supported
+        # Also use --abbrev-commit and --abbrev=0 to display the minimum
+        # number of characters needed per-commit (rather than the full hash)
+        cmd = ['rev-list', '--abbrev-commit', '--abbrev=0', 'HEAD']
+        returncode, stdout, stderr = run_git(cmd)
+        # Fall back on the old method of getting all revisions and counting
+        # the lines
+        if returncode == 0:
+            return str(stdout.count(b'\n'))
+        else:
+            return ''
+    elif sha:
+        return _decode_stdio(stdout)[:40]
+    else:
+        return _decode_stdio(stdout).strip()
+
+
+# Need a recursive glob to find all package data files if there are
+# subdirectories
+import fnmatch
+
+
+def recursive_glob(basedir, pattern):
+    matches = []
+    for root, dirnames, filenames in os.walk(basedir):
+        for filename in fnmatch.filter(filenames, pattern):
+            matches.append(os.path.join(root, filename))
+    return matches
+
+
+# Set affiliated package-specific settings
+PACKAGENAME = 'skyportal_models'
+DESCRIPTION = 'Collaborative Platform for Time-Domain Astronomy'
+LONG_DESCRIPTION = ''
+AUTHOR = 'The Skyportal Developers'
+AUTHOR_EMAIL = ''
+LICENSE = 'BSD'
+URL = 'skyportal.io'
+
+# VERSION should be PEP386 compatible (http://www.python.org/dev/peps/pep-0386)
+VERSION = __version__
+
+# Indicates if this version is a release version
+RELEASE = 'dev' not in VERSION
+
+if not RELEASE:
+    VERSION += get_git_devstr(sha=False)
+
+
+# parse depenency versions from requirements.txt
+deps = ['numpy',
+        'scipy',
+        'astropy',
+        'sqlalchemy-utils',
+        'marshmallow',
+        'marshmallow-sqlalchemy',
+        'marshmallow-enum',
+        'simplejson',
+        'arrow',
+        'tornado',
+        'sncosmo']
+
+
+# these are where the version numbers of the dependencies can be found
+reqfiles = map(lambda f: Path(__file__).parent / f, [
+    'requirements.txt',
+    'baselayer/requirements.txt'
+])
+
+depdict = {}
+for r in reqfiles:
+    with open(r, 'r') as f:
+        for line in f:
+            line = line.strip()
+            m = re.match('([^<=>]+).*', line)
+            packagename = m.group(1)
+            depdict[packagename] = line
+
+# match the dependencies to their version numbers
+deps = [depdict[dep] for dep in deps]
+
+
+setup(name=PACKAGENAME,
+      version=VERSION,
+      description=DESCRIPTION,
+      py_modules=['skyportal.models', 'skyportal.schema',
+                  'skyportal.phot_enum',
+                  'baselayer.__init__',
+                  'baselayer.app.__init__',
+                  'baselayer.app.models',
+                  'baselayer.app.json_util',
+                  'baselayer.app.custom_exceptions'],
+      install_requires=deps,
```

Comment:
`install_requires` can take a list of strings.  Why filter out the version numbers, if they can be used directly?

For packages that use options like, say, `dask[array]`, the square brackets would need to be handled.